### PR TITLE
Adds simple localization features

### PR DESCRIPTION
**Pull request by bitschmidty**
Changes:

- on individual newsletter pages, show available translations, if they exist
- filters on homepage for only english posts
- filters on newsletter page for english posts
- supports language additions
- small fix to newsletter layout html

Notes:

- translations will be located in _posts/[language-code]/newsletters/
- only handles newsletter types for now
- simple language codes (en, ja) for selecting translations. can consider flags or other visuals as well.

Testing:

- add a '_posts/ja/newslettes/2019-10-23-newsletter-ja.md'
- see example ja newsletter here: https://github.com/cryptogarageinc/bitcoinops.github.io/blob/newsletter-69/_posts/ja/newsletters/2019-10-23-newsletter-ja.md

### CODE REVIEW

**Comment by jnewbery** [comment]
ACK 009fd1fbe1339d021c4f802b6bd2e62555c90f39. This seems fine for a low-key rollout of localized newsletters.

If we continue to get japanese and other language translations, we could:

- add flags/graphics/drop-down menu to make this a bit more pleasing on the eye
- add a localized `https://bitcoinops.org/<code>/newsletters/` page so international readers have a place to find all newsletters translated to their language
- add documentation/contrib scripts to make it easier for people to contribute translations.

Thanks for doing this @bitschmidty ! And thanks for handling everything with CG.

**Comment by bitschmidty** [comment]
Rebased and pushed a commit which filters out non english posts from the existing feed.xml

We should also consider adding language specific feeds in the future.

**Comment by bitschmidty** [comment]
Created https://github.com/bitcoinops/bitcoinops.github.io/issues/251 to track non blocking suggestions to localization in the future

**Comment by jnewbery** [comment]
Did you copy the base `feed.xml` from somewhere? (I would have guessed here but it seems like there are some differences). If so, can you split _add feed.xml template which filters for english only_ into two commits:

- the first adds the unaltered `feed.xml` file, so reviewers can use it to generate the feed and verify that there are no differences from master.
- the second makes your modifications to `feed.xml`

**Comment by bitschmidty** [comment]
Split commits for the feed into two:

1. template xml file from 0.10.0 of the jekyll-feed plugin (our current version)
2. tweak to the template file for filtering on english language only

File: newsletters.md [hunk 1]
```diff
@@ -11,9 +11,10 @@ version: 1
 
 {% include newsletter-signup.html %}
 
+{% assign posts_newsletters_en = site.posts | where:"lang","en" | where:"type","newsletter" %}
```
Comment (harding) [inline newsletters.md]: Need to also apply this filtering to the recently-added `/en/publications/` page.

File: _layouts/newsletter.html [hunk 2]
```diff
@@ -2,15 +2,31 @@
...
+{% endfor %}
+
 <article class="newsletter">
   <header class="post-header">
+      {% if localizations.size > 0 %}
+        <a href="{{ '/en' | append:baseurl }}">en</a>
+        {% for locale in localizations %}
+          | <a href="{{ locale.url }}">{{locale.lang}}</a>
+        {% endfor %}
+      {% endif %}
```
Comment (harding) [inline _layouts/newsletter.html]: I think this list would look better right aligned, or maybe center aligned beneath the date; however, I'm ok with any style you like.

**Comment by jnewbery** [comment]
ACK 560573f504bae8c32ef35f1f3b3d96a1417b3316.

Thanks for splitting out the feed change into two commits. I've tested the following:

- with just the first commit (_add jekyll-feed v 0.10.0 feed template file_), there are only whitespace changes
- with the first commit and sample japanese/other language posts, those posts are included in the feed
- with the first commit, sample japanese/other language posts and the second commit (_filter xml feed to english language only_), the feed is the same as master except for whitespace. All non-english posts are filtered out.

Once you've addressed @harding's feedback, this is ready for merge.

File: _layouts/newsletter.html (hunk 2 above)
Comment (jnewbery) [reply _layouts/newsletter.html]: I agree. Let's right-align this.

Comment (bitschmidty) [reply _layouts/newsletter.html]: Right aligned!

File: newsletters.md (hunk 1 above)
Comment (bitschmidty) [reply newsletters.md]: Great catch! Added filtering here.

**Comment by bitschmidty** [comment]
Pushed 2 commits to address feedback from @jnewbery and @harding . Thanks for the review feedback as well as outlining your testing approaches!

**Comment by jnewbery** [comment]
ACK ba22ddd

**Comment by jnewbery** [comment]
I've squashed the two fixup commits into the relevant commits. Merging now.

Thanks for delivering this, @bitschmidty !

**Pull request merged**